### PR TITLE
feat(docs): update SDK pages to reflect new CLI

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Consolidate capability guide.
- Use new local CLI in docs.
- Use React components to collapse bare setup guides (following tracking-transparency format).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/imagepicker.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ import SnackInline from '~/components/plugins/SnackInline';
 
 ## Configuration in app.json / app.config.js
 
-You can configure `expo-image-picker` using its built-in [config plugin](../../../guides/config-plugins.md) if you use config plugins in your project ([EAS Build](../../../build/introduction.md) or `expo run:[android|ios]`). The plugin allows you to configure various properties that cannot be set at runtime and require building a new app binary to take effect.
+You can configure `expo-image-picker` using its built-in [config plugin](../../../guides/config-plugins.md) if you use config plugins in your project ([EAS Build](../../../build/introduction.md) or `npx expo run:[android|ios]`). The plugin allows you to configure various properties that cannot be set at runtime and require building a new app binary to take effect.
```
Comment (Simek) [inline docs/pages/versions/unversioned/sdk/imagepicker.md:27]:
```suggestion
You can configure `expo-image-picker` using its built-in [config plugin](/guides/config-plugins) if you use config plugins in your project ([EAS Build](/build/introduction) or `npx expo run:[android|ios]`). The plugin allows you to configure various properties that cannot be set at runtime and require building a new app binary to take effect.
```

Simplify links.

File: docs/pages/build-reference/ios-capabilities.md [hunk 2]
```diff
@@ -93,6 +93,36 @@ You can run `EXPO_DEBUG=1 eas build` to get more detailed logs regarding the cap
...
+> Preferred method for bare workflow or projects that do **not** using [Expo Prebuild](/workflow/prebuild) to continuously generate the native `ios` and `android` directories.
+
+1. Open the `ios/` directory in Xcode with `xed ios`. If you don't have an `ios/` directory, run `npx expo prebuild -p ios` to generate one.
+2. Then follow [this guide][apple-enable-capability].
+
+### Apple Developer Console
+
+First step is to add the respective key/value pairs to your `ios/[app]/[app].entitlements` (or more specific entitlements file for multi-target apps). You can refer to [Supported Capabilities](#supported-capabilities) to determine which entitlements keys should be added.
+
+1. Log into [Apple Developer Console][apple-dev-console]. Go to "Certificates, Identifiers, & Profiles" > "Identifiers".
```
Comment (Simek) [inline docs/pages/build-reference/ios-capabilities.md:113]:
```suggestion
1. Log into [Apple Developer Console][apple-dev-console]. Click on "Certificates, IDs & Profiles", then navigate to "Identifiers" page.
```

Wording nit. Also, can't we also use link in there? 
* https://developer.apple.com/account/resources/identifiers/list

File: docs/pages/build-reference/ios-capabilities.md [hunk 3]
```diff
@@ -93,6 +93,36 @@ You can run `EXPO_DEBUG=1 eas build` to get more detailed logs regarding the cap
...
+2. Choose the bundle identifier that matches your app's bundle identifier.
+3. Scroll down and enable a capability, some capabilities may require extra setup.
+4. Scroll to the top and press "Save". You will see a dialog that says "Modify App Capabilities", press "Confirm" to continue. You will need to regenerate any provisioning profiles that use this bundle identifier before they'll be valid for building a code signed production ipa.
+
+If the last step is not done correctly then your iOS native build will fail with an error that looks something like:
```
Comment (Simek) [inline docs/pages/build-reference/ios-capabilities.md:118]:
```suggestion
If adding capabilities process has not been done correctly then your iOS native build will fail with an error that looks something like:
```

Generalization & wording.

**Review by Simek — APPROVED** [review]
Great set of updates, LGTM, thank you! 👍 

I have left few minor comments, but there is noting which should block shipping this one.

File: docs/pages/versions/unversioned/sdk/crypto.md [hunk 4]
```diff
@@ -69,7 +69,7 @@ import * as Crypto from 'expo-crypto';
...
-|--------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `ERR_CRYPTO_UNAVAILABLE` | __Web Only.__ Access to the WebCrypto API is restricted to secure origins (https). <br/>You can run your web project from a secure origin with `expo start --https`. |
-| `ERR_CRYPTO_DIGEST`      | An invalid encoding type provided.                                                                                                                              |
+| Code                     | Description                                                                                                                                                              |
+| ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `ERR_CRYPTO_UNAVAILABLE` | **Web Only.** Access to the WebCrypto API is restricted to secure origins (https). <br/>You can run your web project from a secure origin with `npx expo start --https`. |
```
Comment (kbrandwijk) [inline docs/pages/versions/unversioned/sdk/crypto.md:74]: We have pills/labels for this now (web only), I think?

File: docs/pages/versions/unversioned/sdk/document-picker.md [hunk 5]
```diff
@@ -19,25 +21,63 @@ Provides access to the system's UI for selecting documents from the available pr
...
-## Configuration
+## Configuration in app.json / app.config.js
+
+You can configure `expo-document-picker` using its built-in [config plugin](/guides/config-plugins) if you use config plugins in your project ([EAS Build](/build/introduction) or `npx expo run:[android|ios]`). The plugin allows you to configure various properties that cannot be set at runtime and require building a new app binary to take effect. If your app does **not** use EAS Build, then you'll need to manually configure the package.
+
+<ConfigReactNative>
+
+Apps that don't use [EAS Build](/build/introduction) and want [iCloud storage features][icloud-entitlement] must [manually configure](/build-reference/ios-capabilities#manual-setup) the [**iCloud service with CloudKit support**](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_icloud-container-environment) for their bundle identifier.
+
+If you enable through the [Apple Developer Console](/build-reference/ios-capabilities#apple-developer-console), then be sure to add the following entitlements in your `ios/[app]/[app].entitlements` file (where `dev.expo.my-app` if your bundle identifier):
```
Comment (kbrandwijk) [inline docs/pages/versions/unversioned/sdk/document-picker.md:32]: If you enable what?

File: docs/pages/build-reference/ios-capabilities.md (hunk 2 above)
Comment (EvanBacon) [reply docs/pages/build-reference/ios-capabilities.md:113]: this is the only link I use for apple dev console, it's on `apple-dev-console`

File: docs/pages/versions/unversioned/sdk/apple-authentication.md [hunk 6]
```diff
@@ -18,42 +19,42 @@ Beginning with iOS 13, any app that includes third-party authentication options
...
 
-This guide assumes [auto capability sync](../../../build-reference/ios-capabilities) is enabled (Default). Otherwise you'll need to manually enable the capabilities for your bundle identifier through the Apple Developer Console.
+<ConfigReactNative>
 
-#### Managed Workflow
+Apps that don't use [EAS Build](/build/introduction) must [manually configure](/build-reference/ios-capabilities#manual-setup) the **Apple Sign In** capability for their bundle identifier.
 
-1. Build with `eas build -p ios`. EAS Build and the config plugin will automatically configure this service with Apple.
+If you enable through the [Apple Developer Console](/build-reference/ios-capabilities#apple-developer-console), then be sure to add the following entitlements in your `ios/[app]/[app].entitlements` file:
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/apple-authentication.md:30]:
```suggestion
If you enable the **Apple Sign In** capability through the [Apple Developer Console](/build-reference/ios-capabilities#apple-developer-console), then be sure to add the following entitlements in your `ios/[app]/[app].entitlements` file:
```

File: docs/pages/versions/unversioned/sdk/document-picker.md (hunk 5 above)
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/document-picker.md:32]:
```suggestion
If you enable the **iCloud** capability through the [Apple Developer Console](/build-reference/ios-capabilities#apple-developer-console), then be sure to add the following entitlements in your `ios/[app]/[app].entitlements` file (where `dev.expo.my-app` if your bundle identifier):
```

File: docs/pages/versions/unversioned/sdk/crypto.md (hunk 4 above)
Comment (Simek) [reply docs/pages/versions/unversioned/sdk/crypto.md:74]: https://github.com/expo/expo/blob/main/docs/ui/components/Tag/PlatformTag.tsx

Comment (kbrandwijk) [reply docs/pages/versions/unversioned/sdk/crypto.md:74]: I guess we can address this in a follow up PR?

Comment (brentvatne) [reply docs/pages/versions/unversioned/sdk/crypto.md:74]: cc @EvanBacon - looks like you missed @kbrandwijk's comment here - should use the platform tag like @Simek linked to

**Pull request merged**